main.tf

```diff
@@ -62,6 +68,8 @@ locals {
   azuread_domain                           = data.azuread_domains.default.domains[0].domain_name
   primary_region                           = var.primary_region != "" ? var.primary_region : var.regions[0]
   secondary_region                         = var.secondary_region != "" ? var.secondary_region : length(var.regions) > 1 ? var.regions[1] : null
+  region_map                               = var.use_region_shortcodes ? module.region_to_short_region[0].mapping : {}
+  short_regions                            = var.use_region_shortcodes ? [ for region in var.regions: local.region_map[region] ] : []
```

Comment:
Is this `short_regions` local actually needed?

I believe the iteration below for `max_short_region_length` could just use `locals.region_map`

Reply:
It's not needed exactly. Short regions was only to pre-calculate the region translation which would throw the error in that translation, rather than in a "more complicated" logic loop in the max_short_region_length calculation.

We can inline it though if you prefer.

Reply:
I would prefer that to reduce the extra variables

Reply:
Incorporated.